Makefile

```
@@ -96,91 +102,100 @@ docker-build-server: build-server test ## Build docker image with the apiserver.
 docker-push-server: ## Push docker image with the apiserver.
 	docker push ${SERVER_IMG}
 
-##@ Deployment
+##@ Deploy
 
-ifndef ignore-not-found
-  ignore-not-found = false
-endif
+.PHONY: kind-cluster
+kind-cluster: kind kind-cluster-cleanup ## Standup a kind cluster
+	$(KIND) create cluster --name ${KIND_CLUSTER_NAME} 
+	$(KIND) export kubeconfig --name ${KIND_CLUSTER_NAME}
 
-.PHONY: install
-install: manifests kustomize ## Install CRDs into the K8s cluster specified in ~/.kube/config.
+.PHONY: kind-cluster-cleanup
+kind-cluster-cleanup: kind ## Delete the kind cluster
+	$(KIND) delete cluster --name ${KIND_CLUSTER_NAME}
+
+.PHONY: kind-load
+kind-load: kind ## Load the built images onto the local cluster 
+	$(KIND) load docker-image $(CONTROLLER_IMG) --name $(KIND_CLUSTER_NAME)
+	$(KIND) load docker-image $(SERVER_IMG) --name $(KIND_CLUSTER_NAME)
+
+.PHONY: install 
+install: docker-build-server docker-build-controller kind-cluster kind-load cert-manager install-manifests deploy wait ## Install local catalogd
+
+.PHONY: install-manifests
+install-manifests: generate kustomize ## Install CRDs into the K8s cluster specified in ~/.kube/config.
 	$(KUSTOMIZE) build config/crd | kubectl apply -f -
 
-.PHONY: uninstall
-uninstall: manifests kustomize ## Uninstall CRDs from the K8s cluster specified in ~/.kube/config. Call with ignore-not-found=true to ignore resource not found errors during deletion.
+.PHONY: uninstall-manifests
+uninstall-manifests: generate kustomize ## Uninstall CRDs from the K8s cluster specified in ~/.kube/config. 
 	$(KUSTOMIZE) build config/crd | kubectl delete --ignore-not-found=$(ignore-not-found) -f -
 
 .PHONY: deploy
-deploy: cert-manager manifests kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
-	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG}
+deploy: ## Deploy CatalogSource controller and ApiServer to the K8s cluster specified in ~/.kube/config.
+	cd config/manager && $(KUSTOMIZE) edit set image controller=${CONTROLLER_IMG}
 	$(KUSTOMIZE) build config/default | kubectl apply -f -
 
-.PHONY: undeploy
-undeploy: ## Undeploy controller from the K8s cluster specified in ~/.kube/config. Call with ignore-not-found=true to ignore resource not found errors during deletion.
+
+
+ifndef ignore-not-found
+  ignore-not-found = true
+endif
+.PHONY: uninstall
+uninstall: ## Undeploy CatalogSource controller and ApiServer from the K8s cluster specified in ~/.kube/config. Call with ignore-not-found=true to ignore resource not found errors during deletion.
```

if `install` creates a kind cluster, I would expect `uninstall` to clean up that cluster. That may mean we make the actual `kubectl delete ...` step in an intermediate target and make `uninstall` pretty much be:
```suggestion
uninstall: <intermediate> kind-cluster-cleanup ## Undeploy CatalogSource controller and ApiServer from the K8s cluster specified in ~/.kube/config. Call with ignore-not-found=true to ignore resource not found errors during deletion.
```

---

I'm torn about this one. I'm imagining myself installing catalogd another project, say rukpak in the same cluster, but wanting to uninstall just one of those. I could modify the target to remove `kind-cluster-cleanup` target for myself, not sure if that's a good experience for others wanting to use that target? 

Rukpak might not be the best example coz oppy being the gathering place all of these multi project installations/uninstallations would be handled in oppy's makefile, but for what is worth the `install` target has kind-cluster-cleanup so that the old cluster is deleted before a new one is created.

---

For me this is all about the local development so all the clusters I am running at any point should be easy to blow away and recreate and I should be relying on any long running clusters. 

I liked @joelanford's suggestion of adding a target the does a sort of re-deploy without blowing anything away since that could be useful but I would *personally* rather just blow the entire kind cluster away and start fresh.

Another thing is, and maybe I am being a bit naive, I don't think there would be much reason to run another process in the local dev cluster other than catalogd when we are developing for catalogd.

Using the example of oppy - If I am testing stuff for oppy I should be using oppy's Makefile and targets for local development and if I see an issue that is occurring in catalogd I would switch over, fix it, and verify it using the catalogd Makefile targets for a more isolated environment. Once it is fixed, I could hop back over to using oppy's Makefile target and pull in my catalogd changes to verify that in oppy there are no changes that need to be made to fix the issue.

In my head it is all about keeping the individual components that don't have dependencies on each other as isolated as possible to speed up the overall development time (i don't want to have to fiddle with getting rukpak or oppy to work if all I am doing at the moment is making changes in catalogd)

All that being said, if you are strongly against this I am fine keeping it that way

---

You know what, let's just blow it out during uninstall. Let's see if as we start using these targets if there's any wish for a different flavor :) 

Updated the targets to have an `undeploy` and a `uninstall` that is `undeploy+kind-cluster-cleanup`